podman runtime support implemented

d1sgr4c3 opened the pull request:
according to https://github.com/a13xp0p0v/kernel-build-containers/issues/31

in this work i implemented a podman image runtime support for kernel-build-containers.
now we can choose image runtime via `-p` or `-d` flags. moreover, this commit introduces
a way to choose runtime image engine without specifying flag `-p` or `-d` directly. bash
scripts written with strict flag logic.

d1sgr4c3 commented:
hello, @a13xp0p0v 

please have a look at my work

i also left some comments to make your review work easier

d1sgr4c3 commented on manage_images.py:
```diff
@@ -86,18 +89,16 @@ def rm(self):
-                print(f'[!] WARNING: Removing the image {self.id} failed, some containers use it')
-            else:
-                rmi_cmd = self.runtime_cmd + ['rmi', '-f', self.id]
-                subprocess.run(rmi_cmd, text=True, check=True)
```
this diff hunk can look wierd

but dont worry, i just replaced `if not full_id:` with an assertion

this `if` wil succeed only if race condition. so i dropped it to reach 100% coverage

d1sgr4c3 commented on test_image_mgmt.sh:
```diff
@@ -131,6 +131,23 @@ test_error_handling(){
+	export PATH="$PATH_BACK"
+	sudo mv /usr/bin/docker /usr/bin/docker_back
+	python3 -m coverage run -a --branch manage_images.py -l
+	python3 -m coverage run -a --branch manage_images.py -l -d && exit 1
```
also this line covers some code which wasnt covered before

now we reach 100% coverage

```
----------------------------------------------------------------------------------------------
/usr/lib/python3/dist-packages/_distutils_hack/__init__.py     100     96     20      1     4%
manage_images.py                                               159      0     62      2    99%
----------------------------------------------------------------------------------------------
TOTAL                                                          259     96     82      3    66%
```

d1sgr4c3 commented:
@a13xp0p0v, please take a look at `README.md `

i edited the readme with minimal changes. please take a look and tell me which of my new features should i show there?

a13xp0p0v commented:
Hello @d1sgr4c3!

That's cool, thank you!

I'll take a closer look at this pull request on some free weekend.

a13xp0p0v commented:
Hi @d1sgr4c3, I have some time to look at your work this weekend.

Could you please rebase this branch onto the fresh `master` and also give me the ability to push commits into it?

Thanks!

d1sgr4c3 commented on build_linux.py:
```diff
@@ -39,14 +39,14 @@ def get_cross_compile_args(arch):
 def finish_building_kernel(out_dir, interrupt, runtime):
     print('Finish building the kernel')
-    finish_container_cmd = ['bash', os.path.dirname(os.path.abspath(__file__)) + '/finish_container.sh']
+    finish_container_cmd = ['bash', os.path.dirname(os.path.abspath(__file__)) + '/finish_container.sh', runtime]
```
order is `$0 podman/docker kill/nokill out_dir`

here we goes with `bash finish_container.sh docker/podman`

d1sgr4c3 commented:
@a13xp0p0v, i tried to do all carefully, make commits atomically and simplify review itertions

what i did:
1. changes `build_linux.py` similarly to `manage_images.py`
2. the same with `start_container.sh` by idea of `finish_container.sh`
3. improved the argument order: `finish_container.sh podman/docker kill/nokill out_dir`
4. rebased on top of my fix #40  (which is on top of master)
5. reworked test absolutely
6. slightly update the readme
